frame/elections-phragmen/src/benchmarking.rs

```diff
@@ -0,0 +1,453 @@
+fn endowed_account<T: Trait>(name: &'static str, index: u32) -> T::AccountId {
+	let account = account::<T>(name, index);
+	let _ = T::Currency::make_free_balance_be(&account, BalanceOf::<T>::max_value());
+	account
+}
+
+/// Account ot lookup type of system trait.
+fn as_lookup<T: Trait>(account: T::AccountId) -> Lookup<T> {
+	T::Lookup::unlookup(account)
+}
+
+/// Get a reasonable amount of stake based on the execution trait's configuration
+fn default_stake<T: Trait>(factor: u32) -> BalanceOf<T> {
+	let factor = BalanceOf::<T>::from(factor);
+	T::Currency::minimum_balance() * factor
+}
+
+/// Add `c` new candidates.
+fn submit_candidates<T: Trait>(c: u32, prefix: &'static str) -> Result<Vec<T::AccountId>, &'static str> {
+	(0..c).map(|i| {
+		let account = endowed_account::<T>(prefix, i);
+		<Elections<T>>::submit_candidacy(RawOrigin::Signed(account.clone()).into())
+			.map_err(|e| "failed to submit candidacy")?;
+		Ok(account)
+	}).collect::<Result<_, _>>()
+}
+
+/// Add `c` new candidates with self vote.
+fn submit_candidates_with_self_vote<T: Trait>(c: u32, prefix: &'static str) -> Result<Vec<T::AccountId>, &'static str> {
+	let candidates = submit_candidates::<T>(c, prefix)?;
+	let stake = default_stake::<T>(BALANCE_FACTOR);
+	let _ = candidates.iter().map(|c|
+		submit_voter::<T>(c.clone(), vec![c.clone()], stake)
+	).collect::<Result<_, _>>()?;
+	Ok(candidates)
+}
+
+
+/// Submit one voter.
+fn submit_voter<T: Trait>(caller: T::AccountId, votes: Vec<T::AccountId>, stake: BalanceOf<T>)
+	-> Result<(), &'static str>
+{
+	<Elections<T>>::vote(RawOrigin::Signed(caller).into(), votes, stake)
+		.map_err(|e| "failed to submit vote")
+}
+
+/// add `n` locks to account `who`.
+fn add_locks<T: Trait>(who: &T::AccountId, n: u8) {
+	for id in 0..n {
+		let lock_id = [id; 8];
+		let locked = 100;
+		let reasons = WithdrawReason::Transfer | WithdrawReason::Reserve;
+		T::Currency::set_lock(lock_id, who, locked.into(), reasons);
+	}
+}
+
+/// Fill the seats of members and runners-up up until `m`. Note that this might include either only
+/// members, or members and runners-up.
+fn fill_seats_up_to<T: Trait>(m: u32) -> Result<Vec<T::AccountId>, &'static str> {
+	let candidates = submit_candidates_with_self_vote::<T>(m, "fill_seats_up_to")?;
+
+	assert_eq!(<Elections<T>>::candidates().len() as u32, m);
+	<Elections<T>>::do_phragmen();
+	assert_eq!(<Elections<T>>::candidates().len(), 0);
+	assert_eq!(
+		<Elections<T>>::members().len() + <Elections<T>>::runners_up().len(),
+		m as usize,
+	);
+	Ok(candidates)
+}
+
+benchmarks! {
+	_ {}
+
+	// -- Signed ones
+	vote {
+		// range of candidates to vote for.
+		let x in 1 .. (MAXIMUM_VOTE as u32);
+		// range of locks that the caller already had. This extrinsic adds a lock.
+		let l in 1 .. MAX_LOCKS;
+
+		// create a bunch of candidates.
+		let all_candidates = submit_candidates::<T>(MAXIMUM_VOTE as u32, "candidates")?;
+
+		let caller = endowed_account::<T>("caller", SEED);
```

Comment:
please be sure this ends up being `account("caller", 0, 0)` when using the function from the benchmarking library